esiclient/v1/node_network.py

```diff
@@ -157,28 +163,37 @@ def take_action(self, parsed_args):
                 "ERROR: Node {0} must be in the active state".format(
                     node.name))
 
-        baremetal_ports = ironic_client.port.list(node=node_uuid, detail=True)
-        has_free_port = False
-        for bp in baremetal_ports:
-            if 'tenant_vif_port_id' not in bp.internal_info:
-                has_free_port = True
-                break
+        if parsed_args.mac_address:
+            bp = ironic_client.port.get_by_address(parsed_args.mac_address)
+            vif_info = {'port_uuid': bp.uuid}
+            mac_string = " on {0}".format(parsed_args.mac_address)
+        else:
+            vif_info = {}
+            mac_string = ""
 
-        if not has_free_port:
-            raise exceptions.CommandError(
-                "ERROR: Node {0} has no free ports".format(node.name))
+            baremetal_ports = ironic_client.port.list(
+                node=node_uuid, detail=True)
+            has_free_port = False
+            for bp in baremetal_ports:
+                if 'tenant_vif_port_id' not in bp.internal_info:
+                    has_free_port = True
+                    break
+
+            if not has_free_port:
+                raise exceptions.CommandError(
+                    "ERROR: Node {0} has no free ports".format(node.name))
 
```

Comment:
It seems like the order of the code was mostly switched around here and edited some. What does that do?

Reply:
By default, Ironic attaches a VIF to an arbitrary baremetal port - think mac address. The previous code reflected this, merely checking that the node had a free mac address to use. Now, since we're adding the ability to specify a mac address, the code essentially becomes:

```
if mac address specified:
    use specified mac address
else:
    check that node has free mac address
````